feat: Depreciation Gamerule (#8724)
endless-sky/endless-sky: data/gamerules.txt

	# ALLOWABLE VALUES: any integer >= 0
	"npc max mining time" 3600
	
	# Ships with the "frugal" personality will only use weapons that consume ammo or fuel while their
	# health is at or below this fraction.
	# DEFAULT: .75 (75 percent)
	# ALLOWABLE VALUES: any value between 0. and 1.
	"universal frugal threshold" 0.75

	# Depreciation related game rules:

> 	# The depreciated value of an item that is age days old is calculated with the following equation:

Reviewer: ```suggestion
	# The depreciated value of an item that is "age" days old is calculated with the following equation:
```

	# Where: "base value" is the undepreciated value of the item;
	# min is "depreciation min";
	# "grace period" is "depreciation grace period";
	# daily is "depreciation daily"; and
	# "max age" is "depreciation max age", and
	# age is at greater than "grace period" and less than "max age".
	# "depreciated value" = "base value" *
	#   (min + (1 - min) * (daily ^ (age - "grace period")) * ("max age" + "grace period" - age) / "max age")

> 	# The lowest proportion of it's full price the value of a ship or outfit can depreciate to.

Reviewer: ```suggestion
	# The lowest value that a ship or outfit can depreciate to relative to its full price.
	# Captured and gifted items start at this value.
```

> 	# By default, a fully depreciated outfit normally worth 1000 credits will sell for 250 credits.

Reviewer: The default is already provided.
```suggestion
```

> 	# DEFAULT: 0.25

Reviewer: ```suggestion
	# DEFAULT: 0.25 (25 percent)
```

	# ALLOWABLE VALUES: any value between 0. and 1.
	"depreciation min" 0.25

> 	# Number of days before a new outfit and ship begin to depreciate.

Reviewer: ```suggestion
	# Number of days before a new outfit or ship will begin to depreciate.
```

	# DEFAULT: 7 (7 days)
	# ALLOWABLE VALUES: any integer >= 0.
	"depreciation grace period" 7

> 	# DEFAULT: .997 (99.7 percent)

Reviewer: ```suggestion
	# The percentage of an item's depreciable value that it maintains day over day.
	# DEFAULT: .997 (99.7 percent)
```

	# ALLOWABLE VALUES: any integer >= 0
	"npc max mining time" 3600
	
	# Ships with the "frugal" personality will only use weapons that consume ammo or fuel while their
	# health is at or below this fraction.
	# DEFAULT: .75 (75 percent)
	# ALLOWABLE VALUES: any value between 0. and 1.
	"universal frugal threshold" 0.75

> 

Reviewer 2: I think this makes for a much more useful comment than what we have going on further down.
```suggestion

	# Depreciation related game rules:

	# The depreciated value of an item that is age days old is calculated with the following equation:
	# Where: "base value" is the undepreciated value of the item;
	# min is "depreciation min";
	# "grace period" is "depreciation grace period";
	# daily is "depreciation daily"; and
	# "max age" is "depreciation max age", and
	# age is at greater than "grace period" and less than "max age".
	# "depreciated value" = "base value" *
	#   (min + (1 - min) * (daily ^ (age - "grace period")) * ("max age" + "grace period" - age) / "max age")

```

> 	# Minimum percentage of a ship or outfit value at maximum depreciation. A value of .25 (25 percent) means
> 	# a fully depreciated outfit that normally sold at 1000 credits will be sold for 250 credits.
> 	# DEFAULT: .25 (25 percent)
> 	# ALLOWABLE VALUES: any value between 0. and 1.
> 	"depreciation min" 0.25

Reviewer 2: It's not a percentage if it ranges from 0 to 1.
```suggestion
	# The lowest proportion of it's full price the value of a ship or outfit can depreciate to.
	# By default, a fully depreciated outfit normally worth 1000 credits will sell for 250 credits.
	# DEFAULT: 0.25
	# ALLOWABLE VALUES: any value between 0. and 1.
	"depreciation min" 0.25
```

> 	# Amount of days before a new outfit and ship begin to depreciate.

Reviewer 2: ```suggestion
	# Number of days before a new outfit and ship begin to depreciate.
```

	# DEFAULT: 7 (7 days)
	# ALLOWABLE VALUES: any integer >= 0.
	"depreciation grace period" 7

	# First factor of depreciation calculation. The second factor is "depreciation max age" below

> 	# This is the Percentage of a ship or outfit value remains per day after the grace period.

Reviewer 2: ```suggestion
	# This is the proportion of a ship or outfit's value that remains per day after the grace period.
```

> 	# This is the Percentage of an ship or outfit value remains per day after the grace period.

Reviewer 3: ```suggestion
	# This is the Percentage of a ship or outfit value remains per day after the grace period.
```
Fixes the content style check failure.

	# The value is then multiplied by previous value on the next day, and repeat.

> 	# So this can be calculated by "remaining daily percent" = "daily percent" ^ (age - "grace period")

Reviewer 2: What does `"remaining daily percent"` mean?
```suggestion
	# So this can be calculated by "remaining daily proportion" = "daily proportion" ^ (age - "grace period")
```

> 	# First factor of depreciation calculation. The second factor is "depreciation max age" below
> 	# This is the proportion of a ship or outfit's value that remains per day after the grace period.
> 	# The value is then multiplied by previous value on the next day, and repeat.
> 	# So this can be calculated by "remaining daily proportion" = "daily proportion" ^ (age - "grace period")
> 	# where age is in days. "remaining daily percent" is then multiplied by result from next factor.

Reviewer 2: I'm not sure this is very useful. Some of it seems actively harmful.
`This is the proportion of a ship or outfit's value that remains per day after the grace period.`
This is just not true. It is almost true, and it gets closer to being true as max age increases, but it is never actually correct. Describing it this way, along with what comes next, seems like it will, at best, confuse readers enough that they will have to go figure it out for themselves, or, I think more likely, leave them with a misconception about how depreciation actually works.
```suggestion
```

	# DEFAULT: .997 (99.7 percent)
	# ALLOWABLE VALUES: any value between 0. and 1.
	"depreciation daily" 0.997

	# Second factor of depreciation calculation.

> 	# Maximum day for fully depreciated ship or outfit in percent.

Reviewer 4: ```suggestion
	# Maximum day for fully depreciated ship or outfit.
```

> 	# Second factor of depreciation calculation.
> 	# Maximum day for fully depreciated ship or outfit.
> 	# This value will automatically include grace period so 1000 days actually means
> 	# 1000 days + 7 days where 7 days is the grace period. After addition of grace period
> 	# this value is then subtracted by current age of the outfit and divided by max age without grace period
> 	# or in equation: "current age percent" = ("max age" - age) / ("max age" - "grace period")
> 	# where "max age" is the value below added with grace period.
> 	# The final value for current depreciation percentage is:
> 	# "min percent" + (1 - "min percent") * "remaining percent" * "current age percent"
> 	# where "remaining percent" is what calculated from "depreciation daily percent" in gamerule above and
> 	# "min percent" is directly from "depreciation min percent" gamerule

Reviewer 2: Again, I don't think this is very useful.
In particular, a reader here needs to know that the max age is the age after the grace period, but they do not need to know how that is handled internally.
```suggestion
	# The age (in days), after the end of the grace period, at which an item will become fully depreciated.
```
